include/pybind11/pybind11.h

```diff
@@ -2611,8 +2611,8 @@ void print(Args &&...args) {
 }
 
 error_already_set::~error_already_set() {
-    if (m_type) {
-        gil_scoped_acquire gil;
+    gil_scoped_acquire gil;
```

Comment:
This is valid to call after err.restore? m_type could be null after .release() is called on it.

Reply:
This goes hand-in-hand with the change in `restore`, in my current version of the code it doesn't `.release()` anymore.

In the back of my mind I have doubts about the lazy evaluation, restore, and error_already_set being copyable. Maybe unfounded, maybe not.